test/test_utils/ec2.py

```diff
@@ -35,11 +35,9 @@ def get_ec2_instance_type(default, processor, disable_p3dn=False):
             f"Aborting EC2 test run. Unrecognized processor type {processor}. "
             f"Please choose from {allowed_processors}"
         )
-    if default == p3dn:
-        raise RuntimeError("Default instance type should never be p3dn.24xlarge")
     instance_type = os.getenv(f"EC2_{processor.upper()}_INSTANCE_TYPE", default)
-    if instance_type == p3dn and disable_p3dn:
-        instance_type = default
+    if (default == p3dn and instance_type != p3dn) or (instance_type == p3dn and disable_p3dn):
```

Comment:
let's keep the safeguard in place - this function is not the right one to use for EFA tests if they are just going to run on one instance type

Reply:
addressed in other PR.